django/thunderstore/repository/utils.py

```diff
@@ -97,3 +97,22 @@ def get_listing(
     if not package_listing:
         raise Http404("No matching package found")
     return package_listing
+
+
+def solve_listing(
+    owner: Union[Team, str, None] = None,
+    name: Union[str, None] = None,
+    community: Union[Community, None] = None,
+):
+    try:
+        listing = get_listing(
+            owner=owner,
+            name=name,
+            community=community,
+        )
+    except Http404:  # Try to find in another community
```

Comment:
Could check if `community` is not already `None` before attempting a re-fetch with it set to `None`, as otherwise this will end up double-querying when a single query would suffice.

Reply:
I'm not sure what would be the best fix for this code wise, so I'll try my luck with raising a ValuError and catching it in the except; to skip the first query. Let me know if this isn't a preferred way to implement this skip.

Reply:
The way you implemented it seems to work fine but it's backwards from what I had in mind, which would be:
```python
def solve_listing(
    owner: Union[Team, str, None] = None,
    name: Union[str, None] = None,
    community: Union[Community, None] = None,
):
    try:
        listing = get_listing(
            owner=owner,
            name=name,
            community=community,
        )
    except Http404:  # Try to find in another community
        # Skip re-querying if we've already queried with community=None
        if community is None:
            raise
        listing = get_listing(
            owner=owner,
            name=name,
        )
    return listing
```

Worth noting this is hastily edited copypaste so double-check it actually works